spec.html

```diff
@@ -616,7 +637,8 @@ <h1>Runtime Semantics: ValidateTypedArray ( _O_ )</h1>
         1. Assert: _O_ has a [[ViewedArrayBuffer]] internal slot.
         1. Let _buffer_ be _O_.[[ViewedArrayBuffer]].
         1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
-        1. <ins>If CheckIntegerIndexedObjectOutOfBounds(_O_) is *true*, throw a *TypeError* exception.</ins>
+        1. <ins>Let _getBufferByteLength_ be MakeIdempotentArrayBufferByteLengthGetter(~SeqCst~).</ins>
+        1. <ins>If CheckIntegerIndexedObjectOutOfBounds(_O_, _getBufferByteLength_) is *true*, throw a *TypeError* exception.</ins>
```

Comment:
should this have a ! or a ? on it?

Reply:
Nope, these are all predicates.

Reply:
What I said is strictly speaking wrong: the AC returned does a shared memory read, which is observable via the memory model. But it can never result in a thrown exception.

Reply:
I'm talking about both. Predicates usually start with "is" or "has" - the naming "make" doesn't tell me it's a predicate, so the `!` seems useful to communicate that it can't return abruptly.

Reply:
Ah I can see the confusion, especially since "getter" here doesn't mean a JS getter, but an abstract closure that returns the right length (as a real number, so it's never supposed to be user code visible).

Reply:
I would expect an AO named “check” to be like es5’s CheckObjectCoercible, which does the throwing. A predicate generally starts with “has”, “is”, “should”, etc - like it’s asking a question. The “make” AO name seems fine but I don’t know if this “making” can return abruptly or not without ? or !